src/lib.rs

```diff
@@ -46,16 +78,11 @@ impl Default for Decoder {
 }
 
 impl Decoder {
-    pub fn new() -> Self {
+    pub fn new_with_settings(settings: &Settings) -> Self {
         unsafe {
-            let mut settings = mem::MaybeUninit::uninit();
             let mut dec = mem::MaybeUninit::uninit();
 
-            dav1d_default_settings(settings.as_mut_ptr());
-
-            let settings = settings.assume_init();
-
-            let ret = dav1d_open(dec.as_mut_ptr(), &settings);
+            let ret = dav1d_open(dec.as_mut_ptr(), &settings.dav1d_settings);
 
             if ret != 0 {
                 panic!("Cannot instantiate the default decoder {}", ret);
```

Comment:
You probably want to return a `Result` here now. With invalid settings, creating the decoder will fail.